guildwars2/guild/general.py

```diff
@@ -174,60 +174,100 @@ class GeneralGuild:
         except discord.Forbidden:
             await ctx.send("Need permission to embed links")
 
-    @guild.command(name="log", usage="<guild name>")
+    @guild.command(name="log", usage="stash/treasury/members <guild name>")
     @commands.cooldown(1, 10, BucketType.user)
-    async def guild_log(self, ctx, *, guild_name=None):
-        """Get log of last 20 entries of stash
+    async def guild_log(self, ctx, log_type, *, guild_name=None):
+        """Get log of last 20 entries of stash/treasury/members
         Required permissions: guilds and in game permissions"""
-        try:
-            guild = await self.get_guild(ctx, guild_name=guild_name)
-            if not guild:
-                raise BadArgument
-            guild_id = guild["id"]
-            guild_name = guild["name"]
-            endpoint = "guild/{0}/log/".format(guild_id)
-            log = await self.call_api(endpoint, ctx.author, ["guilds"])
-        except (IndexError, APINotFound):
-            return await ctx.send("Invalid guild name")
-        except APIForbidden:
+        state = log_type.lower()
+        member_list = ["invited", "joined", "invite_declined"]
+        if state == "stash" or state == "treasury" or state == "members":
+            try:
+                guild = await self.get_guild(ctx, guild_name=guild_name)
+                if not guild:
+                    raise BadArgument
+                guild_id = guild["id"]
+                guild_name = guild["name"]
+                endpoint = "guild/{0}/log/".format(guild_id)
+                log = await self.call_api(endpoint, ctx.author, ["guilds"])
+            except (IndexError, APINotFound):
+                return await ctx.send("Invalid guild name")
+            except APIForbidden:
+                return await ctx.send(
+                    "You don't have enough permissions in game to "
+                    "use this command")
+            except APIError as e:
+                return await self.error_handler(ctx, e)
+        else:
             return await ctx.send(
-                "You don't have enough permissions in game to "
-                "use this command")
-        except APIError as e:
-            return await self.error_handler(ctx, e)
-        data = discord.Embed(description="Stash Log", colour=self.embed_color)
+                "{0.mention}, Please us either 'stash' or 'treasury' as parameter".
+                    format(ctx.author))
```

Comment:
should provide a command syntax example

Reply:
Changed to "return await self.bot.send_cmd_help(ctx)"